salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+                ret['changes']['new'] = [ob['name'] for ob in local_policies]
+        except Exception:
+            raise
+
+        log.info('Finished cleaning up vault policies.')
+
+
+class VaultAuthManager():
+    """Module for managing Vault Authentication Methods
+    """
+
+    def __init__(self):
+      """Initialize Authentication Manager
+      """
+      log.info("Initializing Vault Auth Manager...")
+
+    def get_remote_auth_methods(self, client, ret):
+        """Retrieve authentication methods from remote vault server
+
+        Arguments:
+            client {hvac} -- hvac client
+            ret {dict} -- result from state
+
+        Returns:
+            list -- auth methods
+        """
+        log.info('Retrieving auth methods from Vault...')
+        auth_resp = client.sys.list_auth_methods()
+
+        auth_methods = []
+        try:
+            for auth_method in auth_resp['data']:
+                auth_methods.append(
+                    VaultAuthMethod(
+                        type=auth_resp[auth_method]['type'],
+                        path=(auth_resp[auth_method]["path"]
+                              if 'path' in auth_resp[auth_method] else auth_method),
+                        description=auth_resp[auth_method]["description"],
+                        config=OrderedDict(
+                            sorted(auth_resp[auth_method]["config"].items()))
+                    )
+                )
+        except Exception:
+            raise
+
+        log.info('Finished retrieving auth methods from vault.')
+
+        return auth_methods
+
+    def populate_local_auth_methods(self, configs, ret):
+        """Get auth methods from local config
+
+        Arguments:
+            configs {list} -- auth methods information
+            ret {dict} -- salt state result
+
+        Returns:
+            list -- auth methods
+        """
+        log.info('Populating local auth methods...')
+
+        auth_methods = []
+        try:
+            for auth_method in configs:
+                auth_config = None
+                extra_config = None
+
+                if "auth_config" in auth_method:
+                    auth_config = OrderedDict(
+                        sorted(auth_method["auth_config"].items()))
+
+                if "extra_config" in auth_method:
+                    extra_config = OrderedDict(
+                        sorted(auth_method["extra_config"].items()))
+
+                auth_methods.append(
+                    VaultAuthMethod(
+                        type=auth_method["type"],
+                        path=auth_method["path"],
+                        description=auth_method["description"],
+                        config=OrderedDict(
+                            sorted(auth_method["config"].items())),
+                        auth_config=auth_config,
+                        extra_config=extra_config
+                    )
+                )
+        except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.